src/pointer-analysis/value_set.cpp

```diff
@@ -902,10 +903,22 @@ void value_sett::get_value_set_rec(
   {
     const dynamic_object_exprt &dynamic_object=
       to_dynamic_object_expr(expr);
+    std::string recency_str="";
```

Comment:
You're already creating a prefix string below, why not just conditionally append to this?